library/src/main/java/com/schibsted/spain/barista/BaristaClickActions.kt

```diff
@@ -0,0 +1,84 @@
+package com.schibsted.spain.barista
+
+import android.support.test.espresso.AmbiguousViewMatcherException
+import android.support.test.espresso.Espresso.onView
+import android.support.test.espresso.Espresso.pressBack
+import android.support.test.espresso.NoMatchingViewException
+import android.support.test.espresso.action.ViewActions.click
+import android.support.test.espresso.action.ViewActions.scrollTo
+import android.support.test.espresso.matcher.ViewMatchers.*
+import android.support.v4.widget.NestedScrollView
+import android.view.View
+import android.widget.HorizontalScrollView
+import android.widget.ListView
+import android.widget.ScrollView
+import com.schibsted.spain.barista.custom.DisplayedMatchers.displayedAnd
+import com.schibsted.spain.barista.custom.NestedEnabledScrollToAction.nestedScrollToAction
+import com.schibsted.spain.barista.internal.failurehandler.SpyFailureHandler
+import com.schibsted.spain.barista.internal.failurehandler.description
+import com.schibsted.spain.barista.internal.resourceMatcher
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers.allOf
+import org.hamcrest.Matchers.anyOf
+
+object BaristaClickActions {
```

Comment:
Why `object`? That's the Kotlin way to do a Singleton, isn't it? And why it's important to be a singleton instead of just a class?

Reply:
It's the way of doing static methods in Kotlin, if I'm not wrong.

Reply:
The class approach would be more verbose if all the members are static:

```
object BaristaClickActions {
   // <-- static members
}
```
--->

```
class BaristaClickActions {
   companion object {
        // <-- static members
   }
   // <-- instance members
}
```

In the end objects and companion objects act like static member containers.
The rationale about objects over statics is that the object concept allows you to extend other classes and implement interfaces while plain statics don't.